### PR TITLE
TST: numpy RuntimeWarning with Series.round()

### LINKED ISSUE #14197: BUG: numpy RuntimeWarning with Series.round()
Maybe needs wrapping in `numpy.errstate` (in pandas)?

``` python
>>> import pandas as pd
>>> pd.__version__
u'0.19.0rc1'
>>> pd.Series([pd.np.nan]).round()

Warning (from warnings module):
  File "C:\Program Files\Python27\lib\site-packages\pandas\core\series.py", line 1340
    result = _values_from_object(self).round(decimals)
RuntimeWarning: invalid value encountered in rint
0   NaN
dtype: float64
```

Same for `DataFrame.round()`.

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [x] closes #14197
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ n/a] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_analytics.py [hunk 1]
```diff
@@ -1842,6 +1842,16 @@ def test_numpy_round(self):
         with pytest.raises(ValueError, match=msg):
             np.round(df, decimals=0, out=df)
 
+    @pytest.mark.skipif(
+        PY2 and is_platform_windows(), reason="numpy/numpy#7882")
+    def test_numpy_round_nan(self):
+        # See gh-14197
+        df = Series([1.53, np.nan, 0.06]).to_frame()
+        with np.errstate(invalid='raise'):
```
Comment (jreback) [inline pandas/tests/frame/test_analytics.py]: can u add assert_produces_warning() to these tests asserting that no warning happens

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_analytics.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/tests/frame/test_analytics.py]: i don't think there is any value add. the RuntimeWarning issued by numpy seems to be only issued once and therefore is highly unlikely to be detected by a test unless run in isolation. hence the `with np.errstate(invalid='raise'):` context mangaer is used to convert the `RuntimeWarning: invalid value encountered in rint` to `FloatingPointError: invalid value encountered in rint`. Also `assert_produces_warning` is not thread-safe.

**Comment by simonjayhawkins** [comment]
@gfyoung : Missing-data? what is required?

**Comment by gfyoung** [comment]
@simonjayhawkins : `missing-data` refers to issues that involve missing data / null values.  It has nothing to do with the fact that your PR is missing any information.  Your PR is just fine in that regard.  🙂

**Comment by simonjayhawkins** [comment]
@gfyoung : thanks 😊

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**